pkg/operation/botanist/botanist.go

```diff
@@ -199,17 +206,23 @@ func (b *Botanist) RequiredExtensionsExist() error {
 	return nil
 }
 
-func (b *Botanist) computeRequiredExtensions() map[string]string {
-	requiredExtensions := map[string]string{
-		extensionsv1alpha1.OperatingSystemConfigResource: string(b.Shoot.GetMachineImageName()),
-	}
+func (b *Botanist) computeRequiredExtensions() map[string]sets.String {
+	requiredExtensions := make(map[string]sets.String)
+
+	requiredExtensions[extensionsv1alpha1.OperatingSystemConfigResource] = sets.NewString(string(b.Shoot.GetMachineImageName()))
 
 	if b.Garden.InternalDomain.Provider != gardenv1beta1.DNSUnmanaged {
-		requiredExtensions[dnsv1alpha1.DNSProviderKind] = b.Garden.InternalDomain.Provider
+		if requiredExtensions[dnsv1alpha1.DNSProviderKind] == nil {
+			requiredExtensions[dnsv1alpha1.DNSProviderKind] = sets.NewString()
+		}
+		requiredExtensions[dnsv1alpha1.DNSProviderKind].Insert(b.Garden.InternalDomain.Provider)
 	}
 
 	if b.Shoot.ExternalDomain != nil && b.Shoot.ExternalDomain.Provider != gardenv1beta1.DNSUnmanaged {
-		requiredExtensions[dnsv1alpha1.DNSProviderKind] = b.Shoot.ExternalDomain.Provider
+		if requiredExtensions[dnsv1alpha1.DNSProviderKind] == nil {
+			requiredExtensions[dnsv1alpha1.DNSProviderKind] = sets.NewString()
+		}
+		requiredExtensions[dnsv1alpha1.DNSProviderKind].Insert(b.Shoot.ExternalDomain.Provider)
 	}
```

Comment:
We need to add all the `Extension` resources here (respect the `globallyEnabled` field).